services/job_runner.go

```diff
@@ -160,24 +160,34 @@ func (rm *jobRunner) workerCount() int {
 	return len(rm.workers)
 }
 
-// BeginRun creates a new run if the job is valid and starts the job.
-func BeginRun(
+// BuildRun checks to ensure the given job has not started or ended before
+// creating a new run for the job.
+func BuildRun(
```

Comment:
Throwing an idea out there that eventually we move BuildRun*, EnqueueRun*, and a few other methods that don't actually need things out of the `adapters` or `services` package into store, where more packages can use them without getting circular dependencies. Happy to wait until there's a more pressing need for it, but I think it's cool that now all those functions just rely on `models` and `store`. Much cleaner.

Reply:
👍 I'll add it in another PR